### PR TITLE
Implement [CustomSubsRegister] screen

### LINKED ISSUE #58: Implement [CustomSubsRegister] screen
[Issuehunt badges](https://issuehunt.io/r/dooboolab/whatssub/issues/58)

@jongtaekchoi

- [ ] Create `screen` component by running `dooboo screen CustomSubsRegister`.
- [ ] Create `shared` components of your taste if you think you need it.
- [ ] Create minimal test code (ex. text input editing and results)
- [ ] Create `shared` components of your taste if you think you need it.
- [ ] Write a blog (extra $10) in one of [medium dooboolab](https://medium.com/@dooboolab) or [medium react-native-seoul](https://medium.com/react-native-seoul) if you prefer Korean.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $20.00)

- [hyochan hyochan](https://issuehunt.io/u/hyochan) ($20.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/dooboolab/whatssub/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by JongtaekChoi**
## Description

- Implement [CustomSubsRegister] screen.
- implement onChangeTexts to [ServiceHeader] components
- add shared component [DatePicker]

## Related Issues

Resolve #58

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/whatssub/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,3 +1,3 @@
-const STORYBOOK_START = false;
+const STORYBOOK_START = true;
```
Comment (hyochan) [inline App.tsx]: Please revert this change when you push a `PR`.

**Comment by hyochan** [comment]
@JongtaekChoi Would you mind changing screen name from [RegisterPersonalProduct] to [CustomServiceRegister]? Also, please run `yarn lint` since there are linting errors reported in ci.

File: App.tsx (hunk 1 above)
Comment (JongtaekChoi) [reply App.tsx]: It's my mistake.

**Comment by JongtaekChoi** [comment]
> @JongtaekChoi Would you mind changing screen name from [RegisterPersonalProduct] to [CustomServiceRegister]? Also, please run `yarn lint` since there are linting errors reported in ci.

I don't mind. Sure I will do that.

**Comment by hyochan** [comment]
@JongtaekChoi Also please try `rebase` it.

**Comment by JongtaekChoi** [comment]
> JoZfMQDTn1
> 
> The above behavior is `bit` awkward. I think you can enhance this in the future.

I did because I thought the user can edit the currency unit in this screen. But come to think of it, it looks strange too. Later, I will enhance this.

**Comment by godon019** [comment]
This is already closed PR but there is something I want to discuss about using `doNotWaitFont` props on providers and suggest you the enhanced models

1. adding such parameter to global component such as provider should be *notified* well, since people using provider have errors while snapshot testing would encounter returning `null` error from their snapshot test. and they have to examine the code deeply until they figure it out that it is caused by provider

2. when adding `optional` parameter
```
interface Props {
  theme?: ThemeType;
  children?: any;
  doNotWaitFont?: boolean;
}
```
such feature(adding optional parameter) is recommended not to introduce any breaking changes to non-font-related component.

with example, currently
```
<Provider>
  <ComponentUsingFont />
</Provider>

<Provider doNotWaitFont>
  <ComponentNotUsingFont />
</Provider>
```
this is bad, because devs who is developing `ComponentNotUsingFont` should be aware of `doNotWaitFont` props' usage. otherwise it will cause error

but
```
<Provider waitFont>
  <ComponentUsingFont />
</Provider>

<Provider>
  <ComponentNotUsingFont />
</Provider>
```
this is better, since devs using `ComponentNotUsingFont` doesn't even need to know `waitFont` props.

3. moreover, providing such feature as props to provider seems inappropriate with its purpose
it might be better to serve 'font-related-state' through context so that any relevant component can use it from context
for example
```
<Provider>
  <ComponentUsingFont />
</Provider>

const ComponentUsingFont = () => {
 const { isFontReady } = useContext();
 return isFontReact ? <Something /> : <Loading />
}
```

So with those three suggestion, I assume the third one is enough to mend all the problems.
This is my opinion and suggestion and I am hoping that this would help you
Also if you think differently I am open learn something new, so feel free to discuss about it!

**Comment by JongtaekChoi** [comment]
> This is already closed PR but there is something I want to discuss about using `doNotWaitFont` props on providers and suggest you the enhanced models
> 
> 1. adding such parameter to global component such as provider should be _notified_ well, since people using provider have errors while snapshot testing would encounter returning `null` error from their snapshot test. and they have to examine the code deeply until they figure it out that it is caused by provider
> 2. when adding `optional` parameter
> 
> ```
> interface Props {
>   theme?: ThemeType;
>   children?: any;
>   doNotWaitFont?: boolean;
> }
> ```
> 
> such feature(adding optional parameter) is recommended not to introduce any breaking changes to non-font-related component.
> 
> with example, currently
> 
> ```
> <Provider>
>   <ComponentUsingFont />
> </Provider>
> 
> <Provider doNotWaitFont>
>   <ComponentNotUsingFont />
> </Provider>
> ```
> 
> this is bad, because devs who is developing `ComponentNotUsingFont` should be aware of `doNotWaitFont` props' usage. otherwise it will cause error
> 
> but
> 
> ```
> <Provider waitFont>
>   <ComponentUsingFont />
> </Provider>
> 
> <Provider>
>   <ComponentNotUsingFont />
> </Provider>
> ```
> 
> this is better, since devs using `ComponentNotUsingFont` doesn't even need to know `waitFont` props.
> 
> 1. moreover, providing such feature as props to provider seems inappropriate with its purpose
>    it might be better to serve 'font-related-state' through context so that any relevant component can use it from context
>    for example
> 
> ```
> <Provider>
>   <ComponentUsingFont />
> </Provider>
> 
> const ComponentUsingFont = () => {
>  const { isFontReady } = useContext();
>  return isFontReady ? <Something /> : <Loading />
> }
> ```
> 
> So with those three suggestion, I assume the third one is enough to mend all the problems.
> This is my opinion and suggestion and I am hoping that this would help you
> Also if you think differently I am open learn something new, so feel free to discuss about it!

Thanks for your opinion. I added this property 'doNotWaitFont' for testing, not for ComponentNotUsingFont. I think if using only the third one and 'doNotWaitFont' property doesn't exist, the updated snapshot will be like below.
```
<Provider>
  < Loading />    
</Provider>
```
This is not the result I want. Actually, I want to make the test wait until loading fonts but I couldn't find the way. 
But, I totally agree that the property is weird and it must be notified well to other developers. 
I think it's my fault to edit this code in this PR. It was a rash act.
If you know the right solution, please submit another PR. 
Thank you.

**Comment by godon019** [comment]
To solve the problem fundamentally I added an hoc like this
```tsx
export function withFontLoading<T>(WrappedComponent: React.FC<T>) {
  const hocComponent = (props: T) => {
    const { doNotWaitFont, isFontReady } = useContext(AppContext);
    return doNotWaitFont || isFontReady ? (
      <WrappedComponent {...props} />
    ) : (
      <LoadingFont />
    );
  };

  hocComponent.displayName = `withFontState(${getDisplayName(
    WrappedComponent,
  )})`;

  return hocComponent;
}
```
so any component using font can just add one like of code like below
```
export default withFontLoading(ProductCard);
```
which solves the warnings

and then to solve test problems with third solution I mentioned above, I added `onFontLoad` promise callback function like below
```
rendered = renderer.create(component(props));
await rendered.getInstance().props.onFontLoad;
expect(rendered.toJSON()).toMatchSnapshot();
```
But then every test code have to wait until the font to be loaded.
Which produces huge delays as the test code is getting bigger
Which is not desirable and impractical

So I used your original solution with encapsulating and collocating those redundant codes!
